### PR TITLE
test(gadget_db): cover host matching boundaries

### LINKED ISSUE #1297: test(gadget_db): cover the `host_matches` domain-boundary matching helper
## Summary

`host_matches` in `src/payload/gadget_db.rs` decides whether an allowlisted host matches a CSP script-gadget host pattern. It carries an important security-relevant rule — a **dotted** pattern (`googleapis.com`) must only match on a domain boundary so that `notgoogle.com` does **not** match `google.com`, while a **bare** fragment (`jquery`, `angular`) keeps loose substring matching. This logic is currently untested.

```rust
fn host_matches(host: &str, pattern: &str) -> bool {
    if pattern.contains('.') {
        host == pattern || host.ends_with(&format!(".{pattern}"))
    } else {
        host.contains(pattern)
    }
}
```

## Why this is a good first issue

- Pure function, no I/O, no async — just inputs and a `bool`.
- The behaviour is fully described in the doc comment above the function.
- The module already has a test module at `src/payload/gadget_db/tests.rs`, so you only need to add cases.

## What to do

Add unit tests to `src/payload/gadget_db/tests.rs` covering at least:

- Dotted pattern exact match: `host_matches("googleapis.com", "googleapis.com")` → `true`
- Dotted pattern subdomain match: `host_matches("ajax.googleapis.com", "googleapis.com")` → `true`
- Dotted pattern boundary guard: `host_matches("notgoogleapis.com", "googleapis.com")` → `false`
- Bare fragment substring match: `host_matches("code.jquery.com", "jquery")` → `true`
- Bare fragment non-match: `host_matches("example.com", "jquery")` → `false`

Because the function is private, put the test inside the crate's existing test module (it already has access via `use super::*;`).

Run `just test` (or `cargo test -p dalfox gadget_db`) to verify.

---

한 줄 요약: CSP 스크립트 가젯 호스트 매칭 함수 `host_matches`의 도메인 경계 검증(예: `notgoogle.com`이 `google.com`에 매칭되지 않아야 함)에 대한 단위 테스트를 추가하는 초보자용 이슈입니다.

### PR DESCRIPTION
**Pull request by aetherloop**
## Summary

- add direct tests for exact and subdomain matches of dotted patterns
- cover missing label boundaries and nonterminal suffix spoofing
- document the intentional loose substring behavior for bare fragments

## Testing

- `cargo test -p dalfox gadget_db`
- `cargo test --quiet`

Closes #1297

### CODE REVIEW

**Review by hahwul — APPROVED** [review]
LGTM

**Comment by hahwul** [comment]
@aetherloop 
Thanks for the PR! It's merged :)
